crates/bevy_asset/src/io/file_asset_io.rs

```diff
@@ -199,13 +201,18 @@ pub fn filesystem_watcher_system(asset_server: Res<AssetServer>) {
                 for path in &paths {
                     let Some(set) = watcher.path_map.get(path) else {continue};
                     for to_reload in set {
-                        if !changed.contains(to_reload) {
-                            changed.insert(to_reload);
-                            let _ = asset_server.load_untracked(to_reload.as_path().into(), true);
-                        }
+                        // When an asset is modified, note down the timestamp
+                        changed.insert(to_reload.to_owned(), Instant::now());
                     }
                 }
             }
+
+            // Reload all assets whose last modification was at least 200ms ago
+            for (to_reload, _) in changed.drain_filter(|_, last_modified| {
+                last_modified.elapsed() >= Duration::from_millis(200)
```

Comment:
This needs a comment justifying why it's needed + why this time period was chosen.

Reply:
Added docs. 50ms was chosen arbitrarily.